tests/foreman/virtwho/test_ui_virtwho.py

```diff
@@ -420,3 +427,155 @@ def test_positive_virtwho_configs_widget(session, form_data):
         assert not session.virtwho_configure.search(name)
         session.organization.select("Default Organization")
         session.organization.delete(org_name)
+
+
+@tier2
+def test_positive_delete_config_delete_user(session, form_data):
+    """Verify when a config is deleted the associated user is deleted.
+
+    :id: 9fdee7f2-833c-47e0-9d58-cd0c9fdd15fe
+
+    :steps:
+        1. Create a virt-who configuration and deploy it to a
+           virt-who server.
+        2. Delete the configuration on the Satellite.
+
+    :expectedresults:
+        1. Verify the virt-who server can no longer connect to the
+           Satellite.
+
+    """
+    name = gen_string('alpha')
+    form_data['name'] = name
+    with session:
+        session.virtwho_configure.create(form_data)
+        config_id = get_configure_id(name)
+        config_command = get_configure_command(config_id)
+        deploy_configure_by_command(config_command)
+        assert session.virtwho_configure.search(name)[0]['Status'] == 'ok'
+        session.virtwho_configure.delete(name)
+        assert not session.virtwho_configure.search(name)
+        restart_virt_who_service()
+        try:
+            deploy_validation()
+        except Exception as Error:
+            assert str(Error) == 'Failed to start virt-who service'
```

Comment:
It's more better if improving `get_virtwho_status`, make it return the service status and error status for rhsm.log.
such as:
```
def get_virtwho_status():
    """Return the status of virt-who service, it will help us to know
    the virt-who configure file is deployed or not.
    """
    _, logs = runcmd('cat /var/log/rhsm/rhsm.log')
    error = len(re.findall(r'\[.*ERROR.*\]', logs))
    ret, stdout = runcmd('systemctl status virt-who')
    running_stauts = ['is running', 'Active: active (running)']
    stopped_status = ['is stopped', 'Active: inactive (dead)']
    if ret != 0:
        return 'undefined'
    if any(key in stdout for key in running_stauts):
        return 'running'
    elif any(key in stdout for key in stopped_status):
        return 'stopped'
    elif error != 0:
        return 'logerror'
    else:
        return 'undefined'
```

Then you can use it here:
```
restart_virtwho_service()
assert get_virtwho_status == 'logerror'
```